### PR TITLE
Other Updates

### PR DESCRIPTION
**Pull request by Furtif**
* Update POGOProtos to v2.12.0 https://github.com/Necrobot-Private/NecroBot/pull/1903/commits/eff2922ea04fcff2b5c8395079c31c5199e9af02
* Added Badges, BonusLoot, RaidTickets, TeamBonusLoot, PokemonDataEgg
* Change forms properties for uses new items https://github.com/Necrobot-Private/NecroBot/pull/1903/commits/3c0d9d3714d334676aaba83dd3524394f387d792
* Fixes/added missed text {vars} for pokemon (catch/fail) and fort events into translations
* Added UseFortItemsTask for use items lure into pokestops
* Add new tasks https://github.com/Necrobot-Private/NecroBot/pull/1903/commits/25e0baab30f34a8efe87d710f525fa2778009ebc
* Added GetGymBadgeDetailsTask https://github.com/Necrobot-Private/NecroBot/pull/1903/commits/f9b24924956a2d1a6224b52317d56b7eab527f5a
*  Added GymFeedPokemon and SfidaRegistration https://github.com/Necrobot-Private/NecroBot/pull/1903/commits/5de928a8d0ab959fc4ce6a994d8d223e0fb66980
* Added GymFeedPokemonTask https://github.com/Necrobot-Private/NecroBot/pull/1903/commits/7fe73617100e352a50d47d6a888b10706e630231
* Fix Names conflits PoGo.Necrobot.Window -> PoGo.NecroBot.Window
Change of procedures in their files
Added Features for new items into Window WPF
CleanUp UnUsed codes https://github.com/Necrobot-Private/NecroBot/pull/1903/commits/85ccb5e01d7fffc8c46f6f1163a2a1ae46659971
* Normal load pokestops (bug update forts) (rb)
* Added ListAvatarCustomizations https://github.com/Necrobot-Private/NecroBot/pull/1903/commits/d5833ab84f385f1bf9c851cb4a322d68c8fddb6f
* Enable use items rare candy and move reroll  https://github.com/Necrobot-Private/NecroBot/pull/1903/commits/7be9d33fb0e94f930ee38bcf7ab4d54f36a9e17c
* Base const data https://github.com/Necrobot-Private/NecroBot/pull/1903/commits/56f9bc5daab5d23170e1b8b3511088f6e1e00c13

### CODE REVIEW

**Comment by Furtif** [comment]
sans_titre

**Comment by Furtif** [comment]
sans_titre

**Comment by Lord-Haji** [comment]
Please update Submodules

**Comment by Lord-Haji** [comment]
Why changing all translations to English?

**Comment by Furtif** [comment]
@Lord-Haji not change it to read complet message only one line changed

**Comment by Furtif** [comment]
sans_titre

**Comment by Furtif** [comment]
https://github.com/Necrobot-Private/PokemonGo.RocketAPI/pull/148

File: PoGo.NecroBot.CLI/Config/Translations/translation.ja-JP.json [hunk 1]
```diff
@@ -78,7 +78,7 @@
     },
     {
       "Key": "eventFortUsed",
-      "Value": "名称: {0} 経験値: {1}, ジェム: {2}, アイテム: {3}, 経度: {4}, 緯度: {5}"
```
Comment (DeathlyPlays) [inline PoGo.NecroBot.CLI/Config/Translations/translation.ja-JP.json]: This is the Japanese Translation for this, like all the others why are you translating them....

File: RocketBot2/Forms/PokeDexForm.cs [hunk 2]
```diff
@@ -27,12 +29,26 @@ public PokeDexForm(Session session, ItemData item = null)
...
+
+            List<PokemonId> list = new List<PokemonId>();
+ 
+            foreach (var pokemon in pokemons)
             {
-                if (e == PokemonId.Missingno || (int)e > 251) continue;
-                Image img = ResourceHelper.SetImageSize(ResourceHelper.GetPokemonImageById((int)e), 48, 48);
-                var Pok = new ItemBox(session, item, e, img);
-                flpPokeDex.Controls.Add(Pok);
+                if (!list.Contains(pokemon.PokemonId))
```
Comment (DeathlyPlays) [inline RocketBot2/Forms/PokeDexForm.cs]: Why isn't the bracket on this line

File: RocketBot2/Forms/PokeDexForm.cs [hunk 3]
```diff
@@ -56,11 +72,12 @@ private void LoadPokemons(ItemData item)
                 Image img = ResourceHelper.SetImageSize(ResourceHelper.GetPokemonImageById((int)e), 48, 48);
                 int vu = 0;
                 int cap = 0;
+                string name = session.Translation.GetPokemonTranslation(e);
                 if (seen.ContainsKey(e)) vu = seen[e];
                 if (cath.ContainsKey(e)) cap = cath[e];
                 if (vu == 0 && cap == 0) img = ResourceHelper.ConvertToBlack(img);
                 if (vu > 0 && cap == 0) img = ResourceHelper.ConvertToBlackAndWhite(img);
-                var Pok = new ItemBox(vu, cap, img);
+                var Pok = new ItemBox(vu, cap, img, name);
```
Comment (DeathlyPlays) [inline RocketBot2/Forms/PokeDexForm.cs]: ``var`` I heard you were converting those right?

File: RocketBot2/Forms/PokeDexForm.cs [hunk 4]
```diff
@@ -27,12 +29,26 @@ public PokeDexForm(Session session, ItemData item = null)
 
         private void LoadPokemons(ItemData item)
         {
-            foreach (PokemonId e in Enum.GetValues(typeof(PokemonId)))
+            var pokemons =
```
Comment (DeathlyPlays) [inline RocketBot2/Forms/PokeDexForm.cs]: Other var

File: RocketBot2/Forms/PokeDexForm.cs [hunk 5]
```diff
@@ -27,12 +29,26 @@ public PokeDexForm(Session session, ItemData item = null)
...
+                  session.Inventory.GetPokemons().Result
+                  .Where(p => p != null && p.PokemonId > 0)
+                  .OrderByDescending(PokemonInfo.CalculatePokemonPerfection)
+                  .ThenByDescending(key => key.Cp)
+                  .OrderBy(key => key.PokemonId);
+
+            List<PokemonId> list = new List<PokemonId>();
+ 
+            foreach (var pokemon in pokemons)
```
Comment (DeathlyPlays) [inline RocketBot2/Forms/PokeDexForm.cs]: var

File: RocketBot2/Forms/MainForm.cs [hunk 6]
```diff
@@ -308,11 +333,13 @@ private void GMAPSatellite_CheckedChanged(object sender, EventArgs e)
...
                     {
-                        Stroke = new Pen(Color.FromArgb(128, 0, 179, 253), 4)
+                        Stroke = new Pen(Color.FromArgb(102, 178, 255), 3)
                     };
                     _pokestopsOverlay.Routes.Add(route);
                 }
 
+                _pokestopsOverlay.Markers.Clear();
+
                 foreach (var pokeStop in pokeStops)
```
Comment (DeathlyPlays) [inline RocketBot2/Forms/MainForm.cs]: var

File: RocketBot2/Forms/MainForm.cs [hunk 7]
```diff
@@ -308,11 +333,13 @@ private void GMAPSatellite_CheckedChanged(object sender, EventArgs e)
                 {
                     var route = new GMapRoute(_routePoints, "Walking Path")
```
Comment (DeathlyPlays) [inline RocketBot2/Forms/MainForm.cs]: var

File: RocketBot2/Forms/MainForm.cs [hunk 8]
```diff
@@ -292,13 +293,37 @@ private void GMAPSatellite_CheckedChanged(object sender, EventArgs e)
...
 
+                try
+                {
+                    if (_session.Navigation.WalkStrategy.Points.Count > 0 && Points != _session.Navigation.WalkStrategy.Points)
+                    {
+                        Points = _session.Navigation.WalkStrategy.Points;
+                        _playerLocations.Clear();
+                        _playerRouteOverlay.Routes.Clear();
+                        List<PointLatLng> routePointLatLngs = new List<PointLatLng>();
+                        foreach (var item in Points)
```
Comment (DeathlyPlays) [inline RocketBot2/Forms/MainForm.cs]: var

File: RocketBot2/Forms/MainForm.cs [hunk 9]
```diff
@@ -292,13 +293,37 @@ private void GMAPSatellite_CheckedChanged(object sender, EventArgs e)
...
             SynchronizationContext.Post(o =>
             {
-                _playerOverlay.Markers.Clear();
                 if (_pokemonsOverlay.Markers.Count > 8)
                     _pokemonsOverlay.Markers.Clear();
-                _pokestopsOverlay.Routes.Clear();
-                _pokestopsOverlay.Markers.Clear();
-                _playerLocations.Clear();
 
+                try
```
Comment (DeathlyPlays) [inline RocketBot2/Forms/MainForm.cs]: Bracket should be on this line

File: RocketBot2/Forms/MainForm.cs [hunk 10]
```diff
@@ -292,13 +293,37 @@ private void GMAPSatellite_CheckedChanged(object sender, EventArgs e)
...
+                {
+                    if (_session.Navigation.WalkStrategy.Points.Count > 0 && Points != _session.Navigation.WalkStrategy.Points)
```
Comment (DeathlyPlays) [inline RocketBot2/Forms/MainForm.cs]: Bracket should be on this line

File: RocketBot2/Forms/MainForm.cs (hunk 8 above)
Comment (DeathlyPlays) [inline RocketBot2/Forms/MainForm.cs]: Also end bracket should be on this line

File: RocketBot2/Forms/ItemBox.cs [hunk 11]
```diff
@@ -12,67 +12,92 @@
...
+                default:
+                    Logger.Write($"{item.ItemId} Can not be used for the moment, the bot still does not completely generate this process.", LogLevel.Warning);
                     break;
             }
-            */
+            PokeDexForm.ActiveForm.Close();
         }
 
-        public ItemBox(int see, int cath, Image pic)
+        public ItemBox(int see, int cath, Image pic, string name)
```
Comment (DeathlyPlays) [inline RocketBot2/Forms/ItemBox.cs]: Move bracket on this line

**Comment by DeathlyPlays** [comment]
That's just a few I saw

**Comment by Furtif** [comment]
?!

**Comment by Furtif** [comment]
ops bad button

File: PoGo.NecroBot.CLI/Config/Translations/translation.ja-JP.json (hunk 1 above)
Comment (Furtif) [inline PoGo.NecroBot.CLI/Config/Translations/translation.ja-JP.json]: This is incomplet look TEMPLATE translation for all vars tranlasion available if it missed into console you no have log complet

File: RocketBot2/Forms/PokeDexForm.cs (hunk 4 above)
Comment (Furtif) [inline RocketBot2/Forms/PokeDexForm.cs]: it is a rocketbot for use item rare or item attaks i uses pokedex form for this

File: RocketBot2/Forms/PokeDexForm.cs (hunk 5 above)
Comment (Furtif) [inline RocketBot2/Forms/PokeDexForm.cs]: same

File: RocketBot2/Forms/PokeDexForm.cs (hunk 2 above)
Comment (Furtif) [inline RocketBot2/Forms/PokeDexForm.cs]: use item rare or othe canot load all pokemons only pokemons into inventory

File: RocketBot2/Forms/PokeDexForm.cs (hunk 3 above)
Comment (Furtif) [inline RocketBot2/Forms/PokeDexForm.cs]: it is for name pokemon into pokedex form 
ditto
see 1
catch 1

File: RocketBot2/Forms/MainForm.cs (hunk 9 above)
Comment (Furtif) [inline RocketBot2/Forms/MainForm.cs]: ???

File: RocketBot2/Forms/MainForm.cs (hunk 10 above)
Comment (Furtif) [inline RocketBot2/Forms/MainForm.cs]: it is for load route point A to point B
sans_titre

File: RocketBot2/Forms/MainForm.cs (hunk 8 above)
Comment (Furtif) [inline RocketBot2/Forms/MainForm.cs]: same

File: RocketBot2/Forms/MainForm.cs (hunk 7 above)
Comment (Furtif) [inline RocketBot2/Forms/MainForm.cs]: same

File: RocketBot2/Forms/MainForm.cs (hunk 6 above)
Comment (Furtif) [inline RocketBot2/Forms/MainForm.cs]: same

File: RocketBot2/Forms/ItemBox.cs (hunk 11 above)
Comment (Furtif) [inline RocketBot2/Forms/ItemBox.cs]: name pokemon translated

File: RocketBot2/Forms/PokeDexForm.cs (hunk 3 above)
Comment (DeathlyPlays) [inline RocketBot2/Forms/PokeDexForm.cs]: no you are still using ``var``

File: RocketBot2/Forms/PokeDexForm.cs (hunk 2 above)
Comment (Furtif) [inline RocketBot2/Forms/PokeDexForm.cs]: image

File: RocketBot2/Forms/PokeDexForm.cs (hunk 3 above)
Comment (Furtif) [inline RocketBot2/Forms/PokeDexForm.cs]: image

File: RocketBot2/Forms/PokeDexForm.cs (hunk 4 above)
Comment (Furtif) [inline RocketBot2/Forms/PokeDexForm.cs]: image

File: RocketBot2/Forms/PokeDexForm.cs (hunk 3 above)
Comment (Furtif) [inline RocketBot2/Forms/PokeDexForm.cs]: image

Comment (Furtif) [inline RocketBot2/Forms/PokeDexForm.cs]: is 3 screnns diferents into same form (pokedexform)

**Comment by Lord-Haji** [comment]
@DeathlyPlays standard C# indentation is not the same as eslint so remove your comments.
And don't review unnecessary stuff

**Comment by Furtif** [comment]
image

**Pull request merged**